packages/replay/src/coreHandlers/handleGlobalEvent.ts

```diff
@@ -0,0 +1,74 @@
+import { Event } from '@sentry/types';
+
+import { REPLAY_EVENT_NAME, UNABLE_TO_SEND_REPLAY } from '../constants';
+import type { ReplayContainer } from '../replay';
+import { addInternalBreadcrumb } from '../util/addInternalBreadcrumb';
+
+/**
+ * Returns a listener to be added to `addGlobalEventProcessor(listener)`.
+ */
+export function handleGlobalEventListener(replay: ReplayContainer): (event: Event) => Event {
+  return (event: Event) => {
+    // Do not apply replayId to the root event
+    if (
+      // @ts-ignore new event type
+      event.type === REPLAY_EVENT_NAME
```

Comment:
m: can we create a issue or add a todo somewhere that adds this event type to the TS types?

Also @JonasBa do we need to do the same for profiling?

Reply:
Yes, this is planned as next step - need to write up an issue!